File: app/src/main/java/net/ralphpina/permissionsmanager/sample/PMApplication.java
package net.ralphpina.permissionsmanager.sample;

import android.app.Application;

import net.ralphpina.permissionsmanager.PermissionsManager;

public class PMApplication extends Application {

    private static PMApplication mInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
        PermissionsManager.init(this);
    }

    public static PMApplication get() {
        return mInstance;
    }
}

File: app/src/main/java/net/ralphpina/permissionsmanager/sample/PMApplication.kt
package net.ralphpina.permissionsmanager.sample

import android.app.Application
import net.ralphpina.permissionsmanager.PermissionsManager
import net.ralphpina.permissionsmanager.android.PermissionsComponent

class PMApplication : Application() {

    lateinit var permissionsManager: PermissionsManager

    override fun onCreate() {
        super.onCreate()
        instance = this
        permissionsManager = PermissionsComponent.Initializer()
                .context(this)
                .prepare()
    }

    companion object {
        lateinit var instance: PMApplication
    }
}
